pkg/oc/cli/cmd/image/append/append.go

```diff
@@ -0,0 +1,656 @@
+		file (the prefix '.wh.' and the filename) which will hide files in the lower layers. All
+		supported filesystem attributes present in the archive will be used as is.
+
+		Metadata about the image (the configuration passed to the container runtime) may be altered
+		by passing a JSON string to the --image or --meta options. The --image flag changes what
+		the container runtime sees, while the --meta option allows you to change the attributes of
+		the image used by the runtime. Use --dry-run to see the result of your changes. You may
+		add the --drop-history flag to remove information from the image about the system that 
+		built the base image.
+
+		Images in manifest list format will automatically select an image that matches the current
+		operating system and architecture unless you use --filter-by-os to select a different image.
+		This flag has no effect on regular images.
+
+		Experimental: This command is under active development and may change without notice.`)
+
+	example = templates.Examples(`
+# Remove the entrypoint on the mysql:latest image
+%[1]s --from mysql:latest --to myregistry.com/myimage:latest --image {"Entrypoint":null}
+
+# Add a new layer to the image
+%[1]s --from mysql:latest --to myregistry.com/myimage:latest layer.tar.gz
+`)
+)
+
+type options struct {
+	Out, ErrOut io.Writer
+
+	From, To   string
+	LayerFiles []string
+
+	ConfigPatch string
+	MetaPatch   string
+
+	DropHistory bool
+	CreatedAt   string
+
+	OSFilter        *regexp.Regexp
+	DefaultOSFilter bool
+
+	FilterByOS string
+
+	MaxPerRegistry int
+
+	DryRun   bool
+	Insecure bool
+	Force    bool
+}
+
+// schema2ManifestOnly specifically requests a manifest list first
+var schema2ManifestOnly = distribution.WithManifestMediaTypes([]string{
+	manifestlist.MediaTypeManifestList,
+	schema2.MediaTypeManifest,
+})
+
+// New creates a new command
+func New(name string, out, errOut io.Writer) *cobra.Command {
+	o := &options{
+		MaxPerRegistry: 3,
+	}
+
+	cmd := &cobra.Command{
+		Use:     "append",
+		Short:   "Add layers to images and push them to a registry",
+		Long:    desc,
+		Example: fmt.Sprintf(example, name),
+		Run: func(c *cobra.Command, args []string) {
+			o.Out = out
+			o.ErrOut = errOut
+			kcmdutil.CheckErr(o.Complete(c, args))
+			kcmdutil.CheckErr(o.Run())
+		},
+	}
+
+	flag := cmd.Flags()
+	flag.BoolVar(&o.DryRun, "dry-run", o.DryRun, "Print the actions that would be taken and exit without writing to the destination.")
+	flag.BoolVar(&o.Insecure, "insecure", o.Insecure, "Allow push and pull operations to registries to be made over HTTP")
+	flag.StringVar(&o.FilterByOS, "filter-by-os", o.FilterByOS, "A regular expression to control which images are mirrored. Images will be passed as '<platform>/<architecture>[/<variant>]'.")
+
+	flag.StringVar(&o.From, "from", o.From, "The image to use as a base. If empty, a new scratch image is created.")
+	flag.StringVar(&o.To, "to", o.To, "The Docker repository tag to upload the appended image to.")
```

Comment:
thoughts on whether this should support imagestreamtags as a "to" target in the future?  (either using the --to flag, or introducing an new flag like --to-imagestreamtag ?)

Reply:
(also --from for that matter)

Reply:
So for both mirror and append I was thinking no, because all clusters should be exposing a public registry URL via the router, and we have oc registry login.  So you should be able to easily get the external name and then use it.

Reply:
Also, you need registry api access anyway to use these commands, so not a burden